### ISSUE TITLE
Improvement of Map Properties

### ISSUE THREAD

**Issue by da-rulez** [AscensionGameDev/Intersect-Engine#721]
**Is your feature request related to a problem? Please describe.**

Right now you have only 2 or 3 (What is Arena) options for map properties. You can either fight, or not. 

**Describe the solution you'd like**
Similar to the bound item enhancements. I would like to see checkboxes for the map properties.

Can walk through others?
Can PvP?
Can attack NPC?
Drop items on death?

Optional: Can attack guild members? - 

**Additional context**

I mention checkboxes, but it might be easier to add this in the server config? That way you can make different presets yourself and they will show up in a dropdown in the editor. This would also make sure existing maps don't get messed up.

So you would have those options in the server config, and you'll be able to add/rename/modify presets there.

**Comment by jcsnider** [comment]
This task absolutely sucks....

First off, I think that all of these options should be individually set per-map. In order to make that bearable we would /someday/ need to add map property presets which could be created/saved/used within the editor anytime. (Long term goals, that is not within the scope of this PR)

That being said it is unacceptable to lose current map settings, the engine would have to look at the current config, and current map types, and set all the individual options properly so behaviors do not change.

The problem is that when we remove the map type options from the config the migration will be unable to read those properties and setup the maps correctly.

The best solution I have is to modify the server options so that upon boot they are loaded into a Newtonsoft JObject (dynamic object), and they are not overwritten until all migrations are complete. And then for migrations which need to see legacy settings they can be referred to via that dynamic JObject, and once all migrations are done the config can be overwritten removing the legacy properties and adding new ones that may exist.

Also marking this as help wanted. Not something I feel like taking on right now.

**Comment by Cheshire92** [comment]
The main issue I see with it is having the legacy options available yeah.
that would basically be a PR of its own, making the legacy options available during a migration.

**Comment by jcsnider** [comment]
Yeah. It's super important that if the migration fails, or the server is sitting at the screen waiting for the user to type in 'READY' that the config isn't overwritten. Once that is solved and we have the legacy options available to query against then this becomes much easier.
